cli/command/deploy_test.go

```diff
@@ -0,0 +1,183 @@
+package command
+
+import (
+	"testing"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+)
+
+func TestCreateDeploy(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewDeployCommand(base.Command())
+
+	file, close := createTempFile(t, "dpl_file")
+	defer close()
+
+	req := models.CreateDeployRequest{
+		DeployName: "dpl_name",
+		DeployFile: []byte("dpl_file"),
+	}
+
+	base.Client.EXPECT().
+		CreateDeploy(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "dpl_id",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	base.Client.EXPECT().
+		ReadDeploy("dpl_id").
+		Return(&models.Deploy{}, nil)
+
+	c := NewContext(t, []string{file.Name(), "dpl_name"}, nil)
+	if err := command.create(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCreateDeploy_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewDeployCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing PATH arg": NewContext(t, nil, nil),
+		"Missing NAME arg": NewContext(t, []string{"path"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := command.create(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestDeleteDeploy(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewDeployCommand(base.Command())
+
+	base.Resolver.EXPECT().
+		Resolve("deploy", "dpl_name").
+		Return([]string{"id"}, nil)
+
+	base.Client.EXPECT().
+		DeleteDeploy("id").
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "dpl_id",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	c := NewContext(t, []string{"dpl_name"}, nil)
+	if err := command.delete(c); err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
Should we be using the [deleteHelper](https://github.com/quintilesims/layer0/blob/cli-refactor/cli/command/base.go#L46) in this function?

Reply:
To be a little more explicit, [this is how I'm using deleteHelper in my PR](https://github.com/quintilesims/layer0/blob/cli-refactor-loadbalancer-command/cli/command/load_balancer.go#L246)

Reply:
I am already using the delete helper in the actual code, this is in `deploy_test.go`